src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -173,6 +191,14 @@ public SocketAsyncContext(int fileDescriptor, SocketAsyncEngine engine)
         {
             _fileDescriptor = fileDescriptor;
             _engine = engine;
+        }
+
+        private void Register()
+        {
+            Debug.Assert(Monitor.IsEntered(_queueLock));
+            Debug.Assert(!_handle.IsAllocated);
+            Debug.Assert(_registeredEvents == SocketAsyncEvents.None);
```

Comment:
Just something to consider: you might want to add messages to all of these Debug.Asserts.  On Unix, we don't get line numbers (at least not yet) for asserts, which means if one of these triggers, without a message it could be difficult to know which failed.